src/component/backdrop/PositioningBackdrop.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {useEffect, useRef, useState} from "react";
+import styles from "./Backdrop.module.scss"
+import {Portal} from "react-portal";
+
+interface PositioningBackdropProps {
+    onDismiss: () => void
+    visible?: boolean
+    children: React.ReactNode
+}
+
+export const PositioningBackdrop: React.FC<PositioningBackdropProps> = (
+    {
+        children,
+        visible = true,
+        onDismiss
+    }
+) => {
+    const boxRef = useRef<HTMLDivElement>(null)
+    const [pos, setPos] = useState({x: 0, y: 0, w: 0, h: 0})
+    const [windowSize, setWindowSize] = useState([
+        window.innerWidth,
+        window.innerHeight,
+    ]);
+
+    useEffect(() => {
+        const handleWindowResize = () => {
+            setWindowSize([window.innerWidth, window.innerHeight]);
+        };
+
+        window.addEventListener('resize', handleWindowResize);
+
+        return () => {
+            window.removeEventListener('resize', handleWindowResize);
+        };
+    });
+
+    useEffect(() => {
+        const ref = boxRef.current
```

Comment:
https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen